clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -588,17 +589,20 @@ private void handleProduceResponse(ClientResponse response, Map<TopicPartition,
     private void completeBatch(ProducerBatch batch, ProduceResponse.PartitionResponse response, long correlationId,
                                long now) {
         Errors error = response.error;
+        String errorMessageSuffix = (response.errorMessage == null || response.errorMessage.isEmpty()) ?
+            "" : String.format(". Error Message: %s", response.errorMessage);
```

Comment:
To address @cmccabe comment, we could replace this by a helper method as he suggested. The method could take the error and the errorMessage to format them appropriately.